packages/dds/tree/src/test/shared-tree/sharedTreeFuzzTests.ts

```diff
@@ -0,0 +1,287 @@
+        };
+        const anchorPath = tree.locate(firstAnchor);
+        assert(compareUpPaths(expectedPath, anchorPath));
+    }
+    return finalState;
+}
+
+function applyFuzzChange(tree: ISharedTree, contents: FuzzChange, transactionResult:TransactionResult): void {
+    const index = contents.index;
+    const nodeField = contents.field;
+    switch (contents.fuzzType) {
+        case "insert":
+            if (index !== undefined && nodeField !== undefined) {
+                tree.runTransaction((forest, editor) => {
+                    const field = editor.sequenceField(contents.parent, nodeField);
+                    field.insert(
+                        index,
+                        singleTextCursor({ type: brand("Test"), value: contents.value }),
+                    );
+                    return transactionResult;
+                });
+            }
+            break;
+        case "delete":
+            if (index !== undefined && nodeField !== undefined) {
+                const parent = contents.parent?.parent;
+                const delField = contents.parent?.parentField;
+                const parentIndex = contents.parent?.parentIndex;
+                if (delField !== undefined && parent !== undefined && parentIndex !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        const field = editor.sequenceField(parent, delField);
+                        field.delete(parentIndex, 1);
+                        return transactionResult;
+                    });
+                }
+            }
+            break;
+        case "setPayload":
+            if (index !== undefined && nodeField !== undefined) {
+                const path = contents.parent;
+                if (path !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        editor.setValue(path, contents.value);
+                        return transactionResult;
+                    });
+                }
+            }
+            break;
+        default:
+            fail("Invalid edit.");
+    }
+}
+
+enum FuzzTestType {
+    Basic = 0,
+    Abort = 1,
+    AnchorStability = 2,
+}
+
+export function runSharedTreeFuzzTests(title: string): void {
+    const random = makeRandom(0)
+    const testBatchSize = 3;
+    describeFuzz(title, ({ testCount }) => {
+        function runFuzzTest (
+            opGenerator: () => AsyncGenerator<Operation, FuzzTestState>,
+            fuzzTestType: FuzzTestType,
+            stepSize: number,
+            batchSize: number
+        ): void {
+            const seed = random.integer(1, 1000000);
+            for (let i = 0; i < batchSize; i++) {
+                const innerRandom = makeRandom(seed);
+                const generatorFactory = () => take(stepSize, opGenerator());
+                switch (fuzzTestType) {
+                    case FuzzTestType.Basic:
+                        it(`with seed ${i}`, async () => {
```

Comment:
The point of having a title that says "with seed x" is that if the test fails, we can easily make a separate test and pass x to it as a seed in order to reproduce the same scenario. That won't work with the current code. I recommend computing a `const runSeed = seed + i;` and passing that to `performFuzzActions` (and putting that runSeed in the title).